// @remotion/cli/dist/browser-download-bar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultBrowserDownloadProgress = void 0;
const renderer_1 = require("@remotion/renderer");
const chalk_1 = require("./chalk");
const log_1 = require("./log");
const make_progress_bar_1 = require("./make-progress-bar");
const progress_bar_1 = require("./progress-bar");
const should_use_non_overlaying_logger_1 = require("./should-use-non-overlaying-logger");
const truthy_1 = require("./truthy");
const makeDownloadProgress = ({ bytesDownloaded, totalBytes, doneIn, }) => {
    const progress = bytesDownloaded / totalBytes;
    return [
        `    +`,
        (0, make_progress_bar_1.makeProgressBar)(progress, false),
        `${doneIn ? 'Got' : 'Getting'} Headless Shell`,
        doneIn === null
            ? (progress * 100).toFixed(0) + '%'
            : chalk_1.chalk.gray(`${doneIn}ms`),
    ]
        .filter(truthy_1.truthy)
        .join(' ');
};
const defaultBrowserDownloadProgress = ({ indent, logLevel, quiet, }) => {
    return () => {
        log_1.Log.info({ indent, logLevel }, 'No local browser could be found.');
        log_1.Log.info({ indent, logLevel }, 'Downloading Chrome Headless Shell https://www.remotion.dev/chrome-headless-shell');
        const updatesDontOverwrite = (0, should_use_non_overlaying_logger_1.shouldUseNonOverlayingLogger)({ logLevel });
        if (updatesDontOverwrite) {
            let lastProgress = 0;
            return {
                version: null,
                onProgress: (progress) => {
                    if (progress.downloadedBytes > lastProgress + 10000000) {
                        lastProgress = progress.downloadedBytes;
                        log_1.Log.info({ indent, logLevel }, `Getting Headless Shell - ${renderer_1.RenderInternals.toMegabytes(progress.downloadedBytes)}/${renderer_1.RenderInternals.toMegabytes(progress.totalSizeInBytes)}`);
                    }
                    if (progress.percent === 1) {
                        log_1.Log.info({ indent, logLevel }, `Got Headless Shell`);
                    }
                },
            };
        }
        const cliOutput = (0, progress_bar_1.createOverwriteableCliOutput)({
            quiet,
            indent,
            cancelSignal: null,
            updatesDontOverwrite,
        });
        const startedAt = Date.now();
        let doneIn = null;
        return {
            version: null,
            onProgress: (progress) => {
                if (progress.percent === 1) {
                    doneIn = Date.now() - startedAt;
                }
                cliOutput.update(makeDownloadProgress({
                    doneIn,
                    bytesDownloaded: progress.downloadedBytes,
                    totalBytes: progress.totalSizeInBytes,
                }), false);
            },
        };
    };
};
exports.defaultBrowserDownloadProgress = defaultBrowserDownloadProgress;
