// @remotion/cli/dist/get-composition-id.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCompositionId = void 0;
const renderer_1 = require("@remotion/renderer");
const studio_server_1 = require("@remotion/studio-server");
const log_1 = require("./log");
const show_compositions_picker_1 = require("./show-compositions-picker");
const getCompName = ({ cliArgs, compositionIdFromUi, }) => {
    if (compositionIdFromUi) {
        return {
            compName: compositionIdFromUi,
            remainingArgs: [],
            reason: 'via UI',
        };
    }
    const [compName, ...remainingArgs] = cliArgs;
    return { compName, remainingArgs, reason: 'Passed as argument' };
};
const getCompositionId = async ({ args, compositionIdFromUi, serializedInputPropsWithCustomSchema, puppeteerInstance, envVariables, timeoutInMilliseconds, chromiumOptions, port, browserExecutable, serveUrlOrWebpackUrl, logLevel, indent, server, offthreadVideoCacheSizeInBytes, binariesDirectory, onBrowserDownload, }) => {
    const { compName, remainingArgs, reason: compReason, } = getCompName({
        cliArgs: args,
        compositionIdFromUi,
    });
    if (compName) {
        const { metadata: config, propsSize } = await renderer_1.RenderInternals.internalSelectComposition({
            id: compName,
            serializedInputPropsWithCustomSchema,
            puppeteerInstance,
            envVariables,
            timeoutInMilliseconds,
            serveUrl: serveUrlOrWebpackUrl,
            browserExecutable,
            chromiumOptions,
            port,
            logLevel,
            server,
            indent,
            onBrowserLog: null,
            offthreadVideoCacheSizeInBytes,
            binariesDirectory,
            onBrowserDownload,
            onServeUrlVisited: () => undefined,
        });
        if (propsSize > 10000000) {
            log_1.Log.warn({
                indent,
                logLevel,
            }, `The props of your composition are large (${studio_server_1.StudioServerInternals.formatBytes(propsSize)}). This may cause slowdown.`);
        }
        if (!config) {
            throw new Error(`Cannot find composition with ID "${compName}"`);
        }
        return {
            compositionId: compName,
            reason: compReason,
            config,
            argsAfterComposition: remainingArgs,
        };
    }
    if (!process.env.CI) {
        const comps = await renderer_1.RenderInternals.internalGetCompositions({
            puppeteerInstance,
            envVariables,
            timeoutInMilliseconds,
            chromiumOptions,
            port,
            browserExecutable,
            logLevel,
            indent,
            server,
            serveUrlOrWebpackUrl,
            onBrowserLog: null,
            serializedInputPropsWithCustomSchema,
            offthreadVideoCacheSizeInBytes,
            binariesDirectory,
            onBrowserDownload,
        });
        const { compositionId, reason } = await (0, show_compositions_picker_1.showSingleCompositionsPicker)(comps, logLevel);
        if (compositionId && typeof compositionId === 'string') {
            return {
                compositionId,
                reason,
                config: comps.find((c) => c.id === compositionId),
                argsAfterComposition: args,
            };
        }
    }
    log_1.Log.error({ indent: false, logLevel }, 'Composition ID not passed.');
    log_1.Log.error({ indent: false, logLevel }, 'Pass an extra argument <composition-id>.');
    process.exit(1);
};
exports.getCompositionId = getCompositionId;
