// @remotion/cli/dist/upgrade.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.upgrade = void 0;
const renderer_1 = require("@remotion/renderer");
const studio_server_1 = require("@remotion/studio-server");
const node_child_process_1 = require("node:child_process");
const node_path_1 = __importDefault(require("node:path"));
const list_of_remotion_packages_1 = require("./list-of-remotion-packages");
const log_1 = require("./log");
const getUpgradeCommand = ({ manager, packages, version, }) => {
    const pkgList = packages.map((p) => `${p}@${version}`);
    const commands = {
        npm: ['i', '--save-exact', '--no-fund', '--no-audit', ...pkgList],
        pnpm: ['i', ...pkgList],
        yarn: ['add', '--exact', ...pkgList],
        bun: ['i', ...pkgList],
    };
    return commands[manager];
};
const upgrade = async (remotionRoot, packageManager, version, logLevel) => {
    var _a, _b, _c;
    const packageJsonFilePath = node_path_1.default.join(remotionRoot, 'package.json');
    const packageJson = require(packageJsonFilePath);
    const dependencies = Object.keys(packageJson.dependencies);
    const devDependencies = Object.keys((_a = packageJson.devDependencies) !== null && _a !== void 0 ? _a : {});
    const optionalDependencies = Object.keys((_b = packageJson.optionalDependencies) !== null && _b !== void 0 ? _b : {});
    const peerDependencies = Object.keys((_c = packageJson.peerDependencies) !== null && _c !== void 0 ? _c : {});
    let targetVersion;
    if (version) {
        targetVersion = version;
        log_1.Log.info({ indent: false, logLevel }, 'Upgrading to specified version: ' + version);
    }
    else {
        targetVersion = await studio_server_1.StudioServerInternals.getLatestRemotionVersion();
        log_1.Log.info({ indent: false, logLevel }, 'Newest Remotion version is', targetVersion);
    }
    const manager = studio_server_1.StudioServerInternals.getPackageManager(remotionRoot, packageManager);
    if (manager === 'unknown') {
        throw new Error(`No lockfile was found in your project (one of ${studio_server_1.StudioServerInternals.lockFilePaths
            .map((p) => p.path)
            .join(', ')}). Install dependencies using your favorite manager!`);
    }
    const toUpgrade = list_of_remotion_packages_1.listOfRemotionPackages.filter((u) => dependencies.includes(u) ||
        devDependencies.includes(u) ||
        optionalDependencies.includes(u) ||
        peerDependencies.includes(u));
    const task = (0, node_child_process_1.spawn)(manager.manager, getUpgradeCommand({
        manager: manager.manager,
        packages: toUpgrade,
        version: targetVersion,
    }), {
        env: {
            ...process.env,
            ADBLOCK: '1',
            DISABLE_OPENCOLLECTIVE: '1',
        },
        stdio: renderer_1.RenderInternals.isEqualOrBelowLogLevel(logLevel, 'info')
            ? 'inherit'
            : 'ignore',
    });
    await new Promise((resolve) => {
        task.on('close', resolve);
    });
    log_1.Log.info({ indent: false, logLevel }, '⏫ Remotion has been upgraded!');
    log_1.Log.info({ indent: false, logLevel }, 'https://remotion.dev/changelog');
};
exports.upgrade = upgrade;
