// @remotion/studio-server/dist/preview-server/routes/can-update-default-props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.canUpdateDefaultPropsHandler = exports.checkIfTypeScriptFile = void 0;
const node_fs_1 = require("node:fs");
const update_default_props_1 = require("../../codemods/update-default-props");
const project_info_1 = require("../project-info");
const checkIfTypeScriptFile = (file) => {
    if (!file.endsWith('.tsx') &&
        !file.endsWith('.ts') &&
        !file.endsWith('.mtsx') &&
        !file.endsWith('.mts')) {
        throw new Error('Cannot update Root file if not using TypeScript');
    }
};
exports.checkIfTypeScriptFile = checkIfTypeScriptFile;
const canUpdateDefaultPropsHandler = async ({ input: { compositionId }, remotionRoot }) => {
    try {
        const projectInfo = await (0, project_info_1.getProjectInfo)(remotionRoot);
        if (!projectInfo.rootFile) {
            throw new Error('Cannot find root file in project');
        }
        (0, exports.checkIfTypeScriptFile)(projectInfo.rootFile);
        await (0, update_default_props_1.updateDefaultProps)({
            compositionId,
            input: (0, node_fs_1.readFileSync)(projectInfo.rootFile, 'utf-8'),
            newDefaultProps: {},
            enumPaths: [],
        });
        return {
            canUpdate: true,
        };
    }
    catch (err) {
        return {
            canUpdate: false,
            reason: err.message,
        };
    }
};
exports.canUpdateDefaultPropsHandler = canUpdateDefaultPropsHandler;
