// @remotion/studio-server/dist/preview-server/routes/update-default-props.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateDefaultPropsHandler = void 0;
const node_fs_1 = require("node:fs");
const update_default_props_1 = require("../../codemods/update-default-props");
const project_info_1 = require("../project-info");
const can_update_default_props_1 = require("./can-update-default-props");
const updateDefaultPropsHandler = async ({ input: { compositionId, defaultProps, enumPaths }, remotionRoot }) => {
    try {
        const projectInfo = await (0, project_info_1.getProjectInfo)(remotionRoot);
        if (!projectInfo.rootFile) {
            throw new Error('Cannot find root file in project');
        }
        (0, can_update_default_props_1.checkIfTypeScriptFile)(projectInfo.rootFile);
        const updated = await (0, update_default_props_1.updateDefaultProps)({
            compositionId,
            input: (0, node_fs_1.readFileSync)(projectInfo.rootFile, 'utf-8'),
            newDefaultProps: JSON.parse(defaultProps),
            enumPaths,
        });
        (0, node_fs_1.writeFileSync)(projectInfo.rootFile, updated);
        return {
            success: true,
        };
    }
    catch (err) {
        return {
            success: false,
            reason: err.message,
            stack: err.stack,
        };
    }
};
exports.updateDefaultPropsHandler = updateDefaultPropsHandler;
