// @remotion/renderer/dist/example-videos.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.exampleVideos = void 0;
const node_path_1 = __importDefault(require("node:path"));
const examplePackage = node_path_1.default.join(__dirname, '..', '..', 'example');
const docsPackage = node_path_1.default.join(__dirname, '..', '..', 'docs');
exports.exampleVideos = {
    bigBuckBunny: node_path_1.default.join(examplePackage, 'public/bigbuckbunny.mp4'),
    transparentWebm: node_path_1.default.join(docsPackage, '/static/img/transparent-video.webm'),
    framerWithoutFileExtension: node_path_1.default.join(examplePackage, 'public', 'framermp4withoutfileextension'),
    corrupted: node_path_1.default.join(examplePackage, 'public', 'corrupted.mp4'),
    customDar: node_path_1.default.join(examplePackage, 'public', 'custom-dar.mp4'),
    screenrecording: node_path_1.default.join(examplePackage, 'public', 'quick.mov'),
    nofps: node_path_1.default.join(examplePackage, 'public', 'nofps.webm'),
    variablefps: node_path_1.default.join(examplePackage, 'public', 'variablefps.webm'),
    zerotimestamp: node_path_1.default.join(examplePackage, 'public', 'zero-timestamp.mp4'),
    webcam: node_path_1.default.join(examplePackage, 'public', 'webcam.webm'),
    iphonevideo: node_path_1.default.join(examplePackage, 'public', 'iphonevideo.mov'),
    av1: node_path_1.default.join(examplePackage, 'public', 'av1.webm'),
    framer24fps: node_path_1.default.join(examplePackage, 'src', 'resources', 'framer-24fps.mp4'),
    music: node_path_1.default.join(examplePackage, 'public', 'music.mp3'),
    notavideo: node_path_1.default.join(examplePackage, 'public', 'giphy.gif'),
    notafile: node_path_1.default.join(examplePackage, 'public', 'doesnotexist'),
    transparentwithdar: node_path_1.default.join(examplePackage, 'public', 'transparent-with-dar.webm'),
    prores: node_path_1.default.join(examplePackage, 'public', 'prores.mov'),
    iphonehevc: node_path_1.default.join(examplePackage, 'public', 'iphone-hevc.mov'),
    matroskaPcm16: node_path_1.default.join(examplePackage, 'public', 'matroska-pcm16.mkv'),
    mp4withmp3: node_path_1.default.join(examplePackage, 'public', 'mp4-mp3.mp4'),
    av1bbb: node_path_1.default.join(examplePackage, 'public', 'av1-bbb.webm'),
    av1mp4: node_path_1.default.join(examplePackage, 'public', 'av1-bbb.mp4'),
    av1mp4WithColr: node_path_1.default.join(examplePackage, 'public', 'av1.mp4'),
    vp8Vorbis: node_path_1.default.join(examplePackage, 'public', 'vp8-vorbis.webm'),
    vp9: node_path_1.default.join(examplePackage, 'public', 'vp9.webm'),
    stretchedVp8: node_path_1.default.join(examplePackage, 'public', 'stretched-vp8.webm'),
    matroskaMp3: node_path_1.default.join(examplePackage, 'public', 'matroska-mp3.mkv'),
    matroskaH265Aac: node_path_1.default.join(examplePackage, 'public', 'matroska-h265-aac.mkv'),
    opusWebm: node_path_1.default.join(examplePackage, 'public', 'opus.webm'),
};
