// @remotion/google-fonts/dist/cjs/Arvo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFont = exports.fontFamily = exports.getInfo = void 0;
const base_1 = require("./base");
const getInfo = () => ({
    fontFamily: 'Arvo',
    importName: 'Arvo',
    version: 'v22',
    url: 'https://fonts.googleapis.com/css2?family=Arvo:ital,wght@0,400;0,700;1,400;1,700',
    unicodeRanges: {
        latin: 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD',
    },
    fonts: {
        italic: {
            '400': {
                latin: 'https://fonts.gstatic.com/s/arvo/v22/tDbN2oWUg0MKqSIg75Tv.woff2',
            },
            '700': {
                latin: 'https://fonts.gstatic.com/s/arvo/v22/tDbO2oWUg0MKqSIoVLH68dr_.woff2',
            },
        },
        normal: {
            '400': {
                latin: 'https://fonts.gstatic.com/s/arvo/v22/tDbD2oWUg0MKqScQ7Q.woff2',
            },
            '700': {
                latin: 'https://fonts.gstatic.com/s/arvo/v22/tDbM2oWUg0MKoZw1-LPK8w.woff2',
            },
        },
    },
});
exports.getInfo = getInfo;
exports.fontFamily = 'Arvo';
const loadFont = (style, options) => {
    return (0, base_1.loadFonts)((0, exports.getInfo)(), style, options);
};
exports.loadFont = loadFont;
