// @remotion/google-fonts/dist/cjs/Economica.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFont = exports.fontFamily = exports.getInfo = void 0;
const base_1 = require("./base");
const getInfo = () => ({
    fontFamily: 'Economica',
    importName: 'Economica',
    version: 'v15',
    url: 'https://fonts.googleapis.com/css2?family=Economica:ital,wght@0,400;0,700;1,400;1,700',
    unicodeRanges: {
        'latin-ext': 'U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF',
        latin: 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD',
    },
    fonts: {
        italic: {
            '400': {
                'latin-ext': 'https://fonts.gstatic.com/s/economica/v15/Qw3ZZQZaHCLgIWa29ZBbM_INKllFfQ.woff2',
                latin: 'https://fonts.gstatic.com/s/economica/v15/Qw3ZZQZaHCLgIWa29ZBbM_IDKlk.woff2',
            },
            '700': {
                'latin-ext': 'https://fonts.gstatic.com/s/economica/v15/Qw3EZQZaHCLgIWa29ZBbM_q4D0x0XHgciw.woff2',
                latin: 'https://fonts.gstatic.com/s/economica/v15/Qw3EZQZaHCLgIWa29ZBbM_q4D0x6XHg.woff2',
            },
        },
        normal: {
            '400': {
                'latin-ext': 'https://fonts.gstatic.com/s/economica/v15/Qw3fZQZaHCLgIWa29ZBbOMIBMl0.woff2',
                latin: 'https://fonts.gstatic.com/s/economica/v15/Qw3fZQZaHCLgIWa29ZBbNsIB.woff2',
            },
            '700': {
                'latin-ext': 'https://fonts.gstatic.com/s/economica/v15/Qw3aZQZaHCLgIWa29ZBTjecUA3x4RHw.woff2',
                latin: 'https://fonts.gstatic.com/s/economica/v15/Qw3aZQZaHCLgIWa29ZBTjecUDXx4.woff2',
            },
        },
    },
});
exports.getInfo = getInfo;
exports.fontFamily = 'Economica';
const loadFont = (style, options) => {
    return (0, base_1.loadFonts)((0, exports.getInfo)(), style, options);
};
exports.loadFont = loadFont;
