// @remotion/google-fonts/dist/cjs/FanwoodText.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFont = exports.fontFamily = exports.getInfo = void 0;
const base_1 = require("./base");
const getInfo = () => ({
    fontFamily: 'Fanwood Text',
    importName: 'FanwoodText',
    version: 'v15',
    url: 'https://fonts.googleapis.com/css2?family=Fanwood+Text:ital,wght@0,400;1,400',
    unicodeRanges: {
        latin: 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD',
    },
    fonts: {
        italic: {
            '400': {
                latin: 'https://fonts.gstatic.com/s/fanwoodtext/v15/3XFzErwl05Ad_vSCF6Fq7xX2R-zb_Pk.woff2',
            },
        },
        normal: {
            '400': {
                latin: 'https://fonts.gstatic.com/s/fanwoodtext/v15/3XFtErwl05Ad_vSCF6Fq7xX2QtzZ.woff2',
            },
        },
    },
});
exports.getInfo = getInfo;
exports.fontFamily = 'Fanwood Text';
const loadFont = (style, options) => {
    return (0, base_1.loadFonts)((0, exports.getInfo)(), style, options);
};
exports.loadFont = loadFont;
