// @remotion/google-fonts/dist/cjs/NotoSansMedefaidrin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFont = exports.fontFamily = exports.getInfo = void 0;
const base_1 = require("./base");
const getInfo = () => ({
    fontFamily: 'Noto Sans Medefaidrin',
    importName: 'NotoSansMedefaidrin',
    version: 'v23',
    url: 'https://fonts.googleapis.com/css2?family=Noto+Sans+Medefaidrin:ital,wght@0,400;0,500;0,600;0,700',
    unicodeRanges: {
        medefaidrin: 'U+16E40-16E9A',
        'latin-ext': 'U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF',
        latin: 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD',
    },
    fonts: {
        normal: {
            '400': {
                medefaidrin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbf5h5YNDIU.woff2',
                'latin-ext': 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfeT99CQs.woff2',
                latin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfdz99.woff2',
            },
            '500': {
                medefaidrin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbf5h5YNDIU.woff2',
                'latin-ext': 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfeT99CQs.woff2',
                latin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfdz99.woff2',
            },
            '600': {
                medefaidrin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbf5h5YNDIU.woff2',
                'latin-ext': 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfeT99CQs.woff2',
                latin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfdz99.woff2',
            },
            '700': {
                medefaidrin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbf5h5YNDIU.woff2',
                'latin-ext': 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfeT99CQs.woff2',
                latin: 'https://fonts.gstatic.com/s/notosansmedefaidrin/v23/WwkAxOq6Dk-wranENynkfeVsNbRZtbOIdLbfdz99.woff2',
            },
        },
    },
});
exports.getInfo = getInfo;
exports.fontFamily = 'Noto Sans Medefaidrin';
const loadFont = (style, options) => {
    return (0, base_1.loadFonts)((0, exports.getInfo)(), style, options);
};
exports.loadFont = loadFont;
