// @remotion/google-fonts/dist/cjs/OdorMeanChey.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadFont = exports.fontFamily = exports.getInfo = void 0;
const base_1 = require("./base");
const getInfo = () => ({
    fontFamily: 'Odor Mean Chey',
    importName: 'OdorMeanChey',
    version: 'v27',
    url: 'https://fonts.googleapis.com/css2?family=Odor+Mean+Chey:ital,wght@0,400',
    unicodeRanges: {
        khmer: 'U+1780-17FF, U+19E0-19FF, U+200C-200D, U+25CC',
        latin: 'U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD',
    },
    fonts: {
        normal: {
            '400': {
                khmer: 'https://fonts.gstatic.com/s/odormeanchey/v27/raxkHiKDttkTe1aOGcJMR1A_4lrU0TukKQ.woff2',
                latin: 'https://fonts.gstatic.com/s/odormeanchey/v27/raxkHiKDttkTe1aOGcJMR1A_4lrf0Ts.woff2',
            },
        },
    },
});
exports.getInfo = getInfo;
exports.fontFamily = 'Odor Mean Chey';
const loadFont = (style, options) => {
    return (0, base_1.loadFonts)((0, exports.getInfo)(), style, options);
};
exports.loadFont = loadFont;
