// @remotion/renderer/dist/browser/BrowserRunner.js
"use strict";
/**
 * Copyright 2020 Google Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _BrowserRunner_executablePath, _BrowserRunner_processArguments, _BrowserRunner_userDataDir, _BrowserRunner_closed, _BrowserRunner_listeners, _BrowserRunner_processClosing;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BrowserRunner = void 0;
const childProcess = __importStar(require("node:child_process"));
const fs = __importStar(require("node:fs"));
const delete_directory_1 = require("../delete-directory");
const logger_1 = require("../logger");
const truthy_1 = require("../truthy");
const Connection_1 = require("./Connection");
const Errors_1 = require("./Errors");
const NodeWebSocketTransport_1 = require("./NodeWebSocketTransport");
const assert_1 = require("./assert");
const should_log_message_1 = require("./should-log-message");
const util_1 = require("./util");
const PROCESS_ERROR_EXPLANATION = `Puppeteer was unable to kill the process which ran the browser binary.
 This means that, on future Puppeteer launches, Puppeteer might not be able to launch the browser.
 Please check your open processes and ensure that the browser processes that Puppeteer launched have been killed.
 If you think this is a bug, please report it on the Puppeteer issue tracker.`;
class BrowserRunner {
    constructor({ executablePath, processArguments, userDataDir, }) {
        _BrowserRunner_executablePath.set(this, void 0);
        _BrowserRunner_processArguments.set(this, void 0);
        _BrowserRunner_userDataDir.set(this, void 0);
        _BrowserRunner_closed.set(this, true);
        _BrowserRunner_listeners.set(this, []);
        _BrowserRunner_processClosing.set(this, void 0);
        __classPrivateFieldSet(this, _BrowserRunner_executablePath, executablePath, "f");
        __classPrivateFieldSet(this, _BrowserRunner_processArguments, processArguments, "f");
        __classPrivateFieldSet(this, _BrowserRunner_userDataDir, userDataDir, "f");
    }
    start(options) {
        var _a, _b;
        const { dumpio, env } = options;
        const stdio = dumpio
            ? ['ignore', 'pipe', 'pipe']
            : ['pipe', 'ignore', 'pipe'];
        (0, assert_1.assert)(!this.proc, 'This process has previously been started.');
        this.proc = childProcess.spawn(__classPrivateFieldGet(this, _BrowserRunner_executablePath, "f"), __classPrivateFieldGet(this, _BrowserRunner_processArguments, "f"), {
            // On non-windows platforms, `detached: true` makes child process a
            // leader of a new process group, making it possible to kill child
            // process tree with `.kill(-pid)` command. @see
            // https://nodejs.org/api/child_process.html#child_process_options_detached
            detached: process.platform !== 'win32',
            env,
            stdio,
        });
        if (dumpio) {
            (_a = this.proc.stdout) === null || _a === void 0 ? void 0 : _a.on('data', (d) => {
                const message = d.toString('utf8').trim();
                if ((0, should_log_message_1.shouldLogBrowserMessage)(message)) {
                    const formatted = (0, should_log_message_1.formatChromeMessage)(message);
                    if (!formatted) {
                        return;
                    }
                    const { output, tag } = formatted;
                    logger_1.Log.verbose({ indent: options.indent, logLevel: options.logLevel, tag }, output);
                }
            });
            (_b = this.proc.stderr) === null || _b === void 0 ? void 0 : _b.on('data', (d) => {
                const message = d.toString('utf8').trim();
                if ((0, should_log_message_1.shouldLogBrowserMessage)(message)) {
                    const formatted = (0, should_log_message_1.formatChromeMessage)(message);
                    if (!formatted) {
                        return;
                    }
                    const { output, tag } = formatted;
                    logger_1.Log.error({ indent: options.indent, logLevel: options.logLevel, tag }, output);
                }
            });
        }
        __classPrivateFieldSet(this, _BrowserRunner_closed, false, "f");
        __classPrivateFieldSet(this, _BrowserRunner_processClosing, new Promise((fulfill, reject) => {
            this.proc.once('exit', () => {
                __classPrivateFieldSet(this, _BrowserRunner_closed, true, "f");
                // Cleanup as processes exit.
                try {
                    if (fs.existsSync(__classPrivateFieldGet(this, _BrowserRunner_userDataDir, "f"))) {
                        (0, delete_directory_1.deleteDirectory)(__classPrivateFieldGet(this, _BrowserRunner_userDataDir, "f"));
                    }
                    fulfill();
                }
                catch (error) {
                    reject(error);
                }
            });
        }), "f");
        __classPrivateFieldSet(this, _BrowserRunner_listeners, [(0, util_1.addEventListener)(process, 'exit', this.kill.bind(this))], "f");
        __classPrivateFieldGet(this, _BrowserRunner_listeners, "f").push((0, util_1.addEventListener)(process, 'SIGINT', () => {
            this.kill();
            process.exit(130);
        }));
        __classPrivateFieldGet(this, _BrowserRunner_listeners, "f").push((0, util_1.addEventListener)(process, 'SIGTERM', this.close.bind(this)));
        __classPrivateFieldGet(this, _BrowserRunner_listeners, "f").push((0, util_1.addEventListener)(process, 'SIGHUP', this.close.bind(this)));
    }
    close() {
        if (__classPrivateFieldGet(this, _BrowserRunner_closed, "f")) {
            return Promise.resolve();
        }
        this.kill();
        // Cleanup this listener last, as that makes sure the full callback runs. If we
        // perform this earlier, then the previous function calls would not happen.
        (0, util_1.removeEventListeners)(__classPrivateFieldGet(this, _BrowserRunner_listeners, "f"));
        return __classPrivateFieldGet(this, _BrowserRunner_processClosing, "f");
    }
    forgetEventLoop() {
        var _a, _b;
        (0, assert_1.assert)(this.proc, 'BrowserRunner not started.');
        this.proc.unref();
        // @ts-expect-error
        (_a = this.proc.stdout) === null || _a === void 0 ? void 0 : _a.unref();
        // @ts-expect-error
        (_b = this.proc.stderr) === null || _b === void 0 ? void 0 : _b.unref();
        (0, assert_1.assert)(this.connection, 'BrowserRunner not connected.');
        this.connection.transport.forgetEventLoop();
    }
    rememberEventLoop() {
        var _a, _b;
        (0, assert_1.assert)(this.proc, 'BrowserRunner not started.');
        this.proc.ref();
        // @ts-expect-error
        (_a = this.proc.stdout) === null || _a === void 0 ? void 0 : _a.ref();
        // @ts-expect-error
        (_b = this.proc.stderr) === null || _b === void 0 ? void 0 : _b.ref();
        (0, assert_1.assert)(this.connection, 'BrowserRunner not connected.');
        this.connection.transport.rememberEventLoop();
    }
    kill() {
        var _a;
        // If the process failed to launch (for example if the browser executable path
        // is invalid), then the process does not get a pid assigned. A call to
        // `proc.kill` would error, as the `pid` to-be-killed can not be found.
        if (((_a = this.proc) === null || _a === void 0 ? void 0 : _a.pid) && pidExists(this.proc.pid)) {
            const { proc } = this;
            try {
                if (process.platform === 'win32') {
                    childProcess.exec(`taskkill /pid ${this.proc.pid} /T /F`, (error) => {
                        if (error) {
                            // taskkill can fail to kill the process e.g. due to missing permissions.
                            // Let's kill the process via Node API. This delays killing of all child
                            // processes of `this.proc` until the main Node.js process dies.
                            proc.kill();
                        }
                    });
                }
                else {
                    // on linux the process group can be killed with the group id prefixed with
                    // a minus sign. The process group id is the group leader's pid.
                    const processGroupId = -this.proc.pid;
                    try {
                        process.kill(processGroupId, 'SIGKILL');
                    }
                    catch (error) {
                        // Killing the process group can fail due e.g. to missing permissions.
                        // Let's kill the process via Node API. This delays killing of all child
                        // processes of `this.proc` until the main Node.js process dies.
                        proc.kill('SIGKILL');
                    }
                }
            }
            catch (error) {
                throw new Error(`${PROCESS_ERROR_EXPLANATION}\nError cause: ${(0, util_1.isErrorLike)(error) ? error.stack : error}`);
            }
        }
        // Attempt to remove temporary profile directory to avoid littering.
        try {
            fs.rmSync(__classPrivateFieldGet(this, _BrowserRunner_userDataDir, "f"), { recursive: true, force: true });
        }
        catch (error) { }
        // Cleanup this listener last, as that makes sure the full callback runs. If we
        // perform this earlier, then the previous function calls would not happen.
        (0, util_1.removeEventListeners)(__classPrivateFieldGet(this, _BrowserRunner_listeners, "f"));
    }
    async setupConnection(options) {
        (0, assert_1.assert)(this.proc, 'BrowserRunner not started.');
        const { timeout } = options;
        const browserWSEndpoint = await waitForWSEndpoint(this.proc, timeout);
        const transport = await NodeWebSocketTransport_1.NodeWebSocketTransport.create(browserWSEndpoint);
        this.connection = new Connection_1.Connection(transport);
        return this.connection;
    }
}
exports.BrowserRunner = BrowserRunner;
_BrowserRunner_executablePath = new WeakMap(), _BrowserRunner_processArguments = new WeakMap(), _BrowserRunner_userDataDir = new WeakMap(), _BrowserRunner_closed = new WeakMap(), _BrowserRunner_listeners = new WeakMap(), _BrowserRunner_processClosing = new WeakMap();
function waitForWSEndpoint(browserProcess, timeout) {
    const browserStderr = browserProcess.stderr;
    (0, assert_1.assert)(browserStderr, '`browserProcess` does not have stderr.');
    let stderrString = '';
    return new Promise((resolve, reject) => {
        browserStderr.addListener('data', onData);
        browserStderr.addListener('close', onClose);
        const listeners = [
            () => browserStderr.removeListener('data', onData),
            () => browserStderr.removeListener('close', onClose),
            (0, util_1.addEventListener)(browserProcess, 'exit', () => {
                return onClose();
            }),
            (0, util_1.addEventListener)(browserProcess, 'error', (error) => {
                return onClose(error);
            }),
        ];
        const timeoutId = timeout ? setTimeout(onTimeout, timeout) : 0;
        function onClose(error) {
            cleanup();
            reject(new Error([
                'Failed to launch the browser process!',
                error ? error.message : null,
                stderrString,
                'Troubleshooting: https://remotion.dev/docs/troubleshooting/browser-launch',
            ]
                .filter(truthy_1.truthy)
                .join('\n')));
        }
        function onTimeout() {
            cleanup();
            reject(new Errors_1.TimeoutError(`Timed out after ${timeout} ms while trying to connect to the browser! Chrome logged the following: ${stderrString}`));
        }
        function onData(data) {
            stderrString += data.toString('utf8');
            const match = stderrString.match(/DevTools listening on (ws:\/\/.*)/);
            if (!match) {
                return;
            }
            cleanup();
            // The RegExp matches, so this will obviously exist.
            resolve(match[1]);
        }
        function cleanup() {
            if (timeoutId) {
                clearTimeout(timeoutId);
            }
            (0, util_1.removeEventListeners)(listeners);
        }
    });
}
function pidExists(pid) {
    try {
        return process.kill(pid, 0);
    }
    catch (error) {
        if ((0, util_1.isErrnoException)(error)) {
            if (error.code && error.code === 'ESRCH') {
                return false;
            }
        }
        throw error;
    }
}
