// @remotion/renderer/dist/browser/Launcher.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChromeLauncher = void 0;
/**
 * Copyright 2017 Google Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const node_fs_1 = __importDefault(require("node:fs"));
const os = __importStar(require("node:os"));
const path = __importStar(require("node:path"));
const Browser_1 = require("./Browser");
const BrowserRunner_1 = require("./BrowserRunner");
const BrowserFetcher_1 = require("./BrowserFetcher");
const tmpDir = () => {
    return process.env.PUPPETEER_TMP_DIR || os.tmpdir();
};
class ChromeLauncher {
    async launch(options) {
        const { args = [], dumpio = false, executablePath, env = process.env, defaultViewport, timeout = 60000, debuggingPort, indent, } = options;
        const chromeArguments = args;
        if (!chromeArguments.some((argument) => {
            return argument.startsWith('--remote-debugging-');
        })) {
            chromeArguments.push(`--remote-debugging-port=${debuggingPort || 0}`);
        }
        // Check for the user data dir argument, which will always be set even
        // with a custom directory specified via the userDataDir option.
        const userDataDir = await node_fs_1.default.promises.mkdtemp(path.join(tmpDir(), 'puppeteer_dev_chrome_profile-'));
        chromeArguments.push(`--user-data-dir=${userDataDir}`);
        let chromeExecutable = executablePath;
        if (!chromeExecutable) {
            const { missingText, executablePath: exPath } = resolveExecutablePath();
            if (missingText) {
                throw new Error(missingText);
            }
            chromeExecutable = exPath;
        }
        const runner = new BrowserRunner_1.BrowserRunner({
            executablePath: chromeExecutable,
            processArguments: chromeArguments,
            userDataDir,
        });
        runner.start({
            dumpio,
            env,
            indent,
            logLevel: options.logLevel,
        });
        let browser;
        try {
            const connection = await runner.setupConnection({
                timeout,
            });
            browser = await Browser_1.HeadlessBrowser._create({
                connection,
                defaultViewport,
                closeCallback: runner.close.bind(runner),
                forgetEventLoop: runner.forgetEventLoop.bind(runner),
                rememberEventLoop: runner.rememberEventLoop.bind(runner),
            });
        }
        catch (error) {
            runner.kill();
            throw error;
        }
        try {
            await browser.waitForTarget((t) => {
                return t.type() === 'page';
            }, { timeout });
        }
        catch (error) {
            await browser.close(false, options.logLevel, options.indent);
            throw error;
        }
        return browser;
    }
    executablePath() {
        const results = resolveExecutablePath();
        return results.executablePath;
    }
}
exports.ChromeLauncher = ChromeLauncher;
function resolveExecutablePath() {
    const revisionInfo = (0, BrowserFetcher_1.getRevisionInfo)();
    const missingText = revisionInfo.local
        ? undefined
        : `Could not find expected browser locally.`;
    return { executablePath: revisionInfo.executablePath, missingText };
}
