// @remotion/google-fonts/dist/esm/KoHo.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/KoHo.ts
var getInfo = () => ({
  fontFamily: "KoHo",
  importName: "KoHo",
  version: "v16",
  url: "https://fonts.googleapis.com/css2?family=KoHo:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;1,200;1,300;1,400;1,500;1,600;1,700",
  unicodeRanges: {
    thai: "U+0E01-0E5B, U+200C-200D, U+25CC",
    vietnamese: "U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB",
    "latin-ext": "U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF",
    latin: "U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD"
  },
  fonts: {
    italic: {
      "200": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNisssJ_DMaqlPys.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNisssJ_DKqqlPys.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNisssJ_DK6qlPys.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNisssJ_DJaql.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNiss1JzDMaqlPys.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNiss1JzDKqqlPys.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNiss1JzDK6qlPys.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNiss1JzDJaql.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FwfZ5fmddNNiska77WGos.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FwfZ5fmddNNiskcL7WGos.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FwfZ5fmddNNiskcb7WGos.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FwfZ5fmddNNiskf77W.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissjJ3DMaqlPys.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissjJ3DKqqlPys.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissjJ3DK6qlPys.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissjJ3DJaql.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissoJrDMaqlPys.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissoJrDKqqlPys.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissoJrDK6qlPys.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissoJrDJaql.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissxJvDMaqlPys.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissxJvDKqqlPys.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissxJvDK6qlPys.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FzfZ5fmddNNissxJvDJaql.woff2"
      }
    },
    normal: {
      "200": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPuE1aI3zJ7Kh.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPuE1aJbzJ7Kh.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPuE1aJfzJ7Kh.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPuE1aJnzJw.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPoU2aI3zJ7Kh.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPoU2aJbzJ7Kh.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPoU2aJfzJ7Kh.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPoU2aJnzJw.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2F-fZ5fmddNNjoUfabS.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2F-fZ5fmddNNiEUfabS.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2F-fZ5fmddNNiAUfabS.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2F-fZ5fmddNNi4UfQ.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPt03aI3zJ7Kh.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPt03aJbzJ7Kh.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPt03aJfzJ7Kh.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPt03aJnzJw.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPvEwaI3zJ7Kh.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPvEwaJbzJ7Kh.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPvEwaJfzJ7Kh.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPvEwaJnzJw.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPpUxaI3zJ7Kh.woff2",
        vietnamese: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPpUxaJbzJ7Kh.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPpUxaJfzJ7Kh.woff2",
        latin: "https://fonts.gstatic.com/s/koho/v16/K2FxfZ5fmddNPpUxaJnzJw.woff2"
      }
    }
  }
});
var fontFamily = "KoHo";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
