// @remotion/google-fonts/dist/esm/Krub.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/Krub.ts
var getInfo = () => ({
  fontFamily: "Krub",
  importName: "Krub",
  version: "v9",
  url: "https://fonts.googleapis.com/css2?family=Krub:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;1,200;1,300;1,400;1,500;1,600;1,700",
  unicodeRanges: {
    thai: "U+0E01-0E5B, U+200C-200D, U+25CC",
    vietnamese: "U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB",
    "latin-ext": "U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF",
    latin: "U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD"
  },
  fonts: {
    italic: {
      "200": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQiwLBCUIASSf8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQiwLBCS4ASSf8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQiwLBCSoASSf8.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQiwLBCRIAS.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQipLNCUIASSf8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQipLNCS4ASSf8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQipLNCSoASSf8.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQipLNCRIAS.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlFdRyC6CRYbkQqG5FXe6E.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlFdRyC6CRYbkQqAJFXe6E.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlFdRyC6CRYbkQqAZFXe6E.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlFdRyC6CRYbkQqD5FX.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi_LJCUIASSf8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi_LJCS4ASSf8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi_LJCSoASSf8.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi_LJCRIAS.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi0LVCUIASSf8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi0LVCS4ASSf8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi0LVCSoASSf8.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQi0LVCRIAS.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQitLRCUIASSf8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQitLRCS4ASSf8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQitLRCSoASSf8.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlGdRyC6CRYbkQitLRCRIAS.woff2"
      }
    },
    normal: {
      "200": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZo47GKJyRpgW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZo47GLlyRpgW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZo47GLhyRpgW.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZo47GLZyRg.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZuo4GKJyRpgW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZuo4GLlyRpgW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZuo4GLhyRpgW.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZuo4GLZyRg.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlLdRyC6CRYblUaDYlT.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlLdRyC6CRYbk4aDYlT.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlLdRyC6CRYbk8aDYlT.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlLdRyC6CRYbkEaDQ.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZrI5GKJyRpgW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZrI5GLlyRpgW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZrI5GLhyRpgW.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZrI5GLZyRg.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZp4-GKJyRpgW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZp4-GLlyRpgW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZp4-GLhyRpgW.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZp4-GLZyRg.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZvo_GKJyRpgW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZvo_GLlyRpgW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZvo_GLhyRpgW.woff2",
        latin: "https://fonts.gstatic.com/s/krub/v9/sZlEdRyC6CRYZvo_GLZyRg.woff2"
      }
    }
  }
});
var fontFamily = "Krub";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
