// @remotion/google-fonts/dist/esm/Linefont.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/Linefont.ts
var getInfo = () => ({
  fontFamily: "Linefont",
  importName: "Linefont",
  version: "v3",
  url: "https://fonts.googleapis.com/css2?family=Linefont:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900",
  unicodeRanges: {},
  fonts: {}
});
var fontFamily = "Linefont";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
