// @remotion/google-fonts/dist/esm/Mali.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/Mali.ts
var getInfo = () => ({
  fontFamily: "Mali",
  importName: "Mali",
  version: "v10",
  url: "https://fonts.googleapis.com/css2?family=Mali:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;1,200;1,300;1,400;1,500;1,600;1,700",
  unicodeRanges: {
    thai: "U+0E01-0E5B, U+200C-200D, U+25CC",
    vietnamese: "U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB",
    "latin-ext": "U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF",
    latin: "U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD"
  },
  fonts: {
    italic: {
      "200": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8wlVgMv7SodY.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8wlVgKf7SodY.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8wlVgKP7SodY.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8wlVgJv7S.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8plZgMv7SodY.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8plZgKf7SodY.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8plZgKP7SodY.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8plZgJv7S.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bU2SRONuN4SCj0GXR1Gd8.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bU2SRONuN4SCj0AnR1Gd8.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bU2SRONuN4SCj0A3R1Gd8.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bU2SRONuN4SCj0DXR1.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8_ldgMv7SodY.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8_ldgKf7SodY.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8_ldgKP7SodY.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8_ldgJv7S.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj80lBgMv7SodY.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj80lBgKf7SodY.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj80lBgKP7SodY.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj80lBgJv7S.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8tlFgMv7SodY.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8tlFgKf7SodY.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8tlFgKP7SodY.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bX2SRONuN4SCj8tlFgJv7S.woff2"
      }
    },
    normal: {
      "200": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QOLlGkdQJObW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QOLlGlxQJObW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QOLlGl1QJObW.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QOLlGlNQJA.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QIbmGkdQJObW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QIbmGlxQJObW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QIbmGl1QJObW.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QIbmGlNQJA.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0ba2SRONuN4SDnED2xx.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0ba2SRONuN4SCLED2xx.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0ba2SRONuN4SCPED2xx.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0ba2SRONuN4SC3EDw.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QN7nGkdQJObW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QN7nGlxQJObW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QN7nGl1QJObW.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QN7nGlNQJA.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QPLgGkdQJObW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QPLgGlxQJObW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QPLgGl1QJObW.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QPLgGlNQJA.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QJbhGkdQJObW.woff2",
        vietnamese: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QJbhGlxQJObW.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QJbhGl1QJObW.woff2",
        latin: "https://fonts.gstatic.com/s/mali/v10/N0bV2SRONuN4QJbhGlNQJA.woff2"
      }
    }
  }
});
var fontFamily = "Mali";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
