// @remotion/google-fonts/dist/esm/NotoSansLao.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/NotoSansLao.ts
var getInfo = () => ({
  fontFamily: "Noto Sans Lao",
  importName: "NotoSansLao",
  version: "v30",
  url: "https://fonts.googleapis.com/css2?family=Noto+Sans+Lao:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900",
  unicodeRanges: {
    lao: "U+0E81-0EDF, U+200C-200D, U+25CC",
    "latin-ext": "U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF",
    latin: "U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD"
  },
  fonts: {
    normal: {
      "100": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "200": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "300": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "400": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "500": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "600": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "700": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "800": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      },
      "900": {
        lao: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg46D6ELWw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg49D6ELWw.woff2",
        latin: "https://fonts.gstatic.com/s/notosanslao/v30/bx6QNx2Ol_ixgdYWLm9BwxM3NW6BOkuf763Clj73Cg4zD6E.woff2"
      }
    }
  }
});
var fontFamily = "Noto Sans Lao";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
