// @remotion/google-fonts/dist/esm/Trirong.mjs
// src/base.ts
import {continueRender, delayRender} from "remotion";
var loadedFonts = {};
var loadFonts = (meta, style, options) => {
  const promises = [];
  const styles = style ? [style] : Object.keys(meta.fonts);
  for (const style2 of styles) {
    if (typeof FontFace === "undefined") {
      continue;
    }
    if (!meta.fonts[style2]) {
      throw new Error(`The font ${meta.fontFamily} does not have a style ${style2}`);
    }
    const weights = options?.weights ?? Object.keys(meta.fonts[style2]);
    for (const weight of weights) {
      if (!meta.fonts[style2][weight]) {
        throw new Error(`The font ${meta.fontFamily} does not  have a weight ${weight} in style ${style2}`);
      }
      const subsets = options?.subsets ?? Object.keys(meta.fonts[style2][weight]);
      for (const subset of subsets) {
        let font = meta.fonts[style2]?.[weight]?.[subset];
        if (!font) {
          throw new Error(`weight: ${weight} subset: ${subset} is not available for '${meta.fontFamily}'`);
        }
        let fontKey = `${meta.fontFamily}-${style2}-${weight}-${subset}`;
        const previousPromise = loadedFonts[fontKey];
        if (previousPromise) {
          promises.push(previousPromise);
          continue;
        }
        const handle = delayRender(`Fetching ${meta.fontFamily} font ${JSON.stringify({
          style: style2,
          weight,
          subset
        })}`);
        const fontFace = new FontFace(meta.fontFamily, `url(${font}) format('woff2')`, {
          weight,
          style: style2,
          unicodeRange: meta.unicodeRanges[subset]
        });
        const promise = fontFace.load().then(() => {
          (options?.document ?? document).fonts.add(fontFace);
          continueRender(handle);
        }).catch((err) => {
          loadedFonts[fontKey] = undefined;
          throw err;
        });
        loadedFonts[fontKey] = promise;
        promises.push(promise);
      }
    }
  }
  return {
    fontFamily: meta.fontFamily,
    fonts: meta.fonts,
    unicodeRanges: meta.unicodeRanges,
    waitUntilDone: () => Promise.all(promises).then(() => {
      return;
    })
  };
};

// src/Trirong.ts
var getInfo = () => ({
  fontFamily: "Trirong",
  importName: "Trirong",
  version: "v15",
  url: "https://fonts.googleapis.com/css2?family=Trirong:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900",
  unicodeRanges: {
    thai: "U+0E01-0E5B, U+200C-200D, U+25CC",
    vietnamese: "U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB",
    "latin-ext": "U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20C0, U+2113, U+2C60-2C7F, U+A720-A7FF",
    latin: "U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD"
  },
  fonts: {
    italic: {
      "100": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3CqXNgp8wxdOdOn44Qub5ywuzHgw.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3CqXNgp8wxdOdOn44Qub5pwuzHgw.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3CqXNgp8wxdOdOn44Qub5owuzHgw.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3CqXNgp8wxdOdOn44Qub5mwuw.woff2"
      },
      "200": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QFa9x8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QFa9x6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QFa9x68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QFa9x5cn6.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qcaxx8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qcaxx6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qcaxx68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qcaxx5cn6.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOn44Yzo5k2ug.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOn44Y1Y5k2ug.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOn44Y1I5k2ug.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOn44Y2o5k.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QKa1x8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QKa1x6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QKa1x68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QKa1x5cn6.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QBapx8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QBapx6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QBapx68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QBapx5cn6.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QYatx8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QYatx6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QYatx68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QYatx5cn6.woff2"
      },
      "800": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qfahx8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qfahx6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qfahx68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44Qfahx5cn6.woff2"
      },
      "900": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QWalx8cn6uts.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QWalx6sn6uts.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QWalx68n6uts.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3BqXNgp8wxdOdOn44QWalx5cn6.woff2"
      }
    },
    normal: {
      "100": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOl-gYzo5k2ug.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOl-gY1Y5k2ug.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOl-gY1I5k2ug.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3EqXNgp8wxdOdOl-gY2o5k.woff2"
      },
      "200": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl0QJzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl0QJzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl0QJzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl0QJzalB5w.woff2"
      },
      "300": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlyAKzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlyAKzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlyAKzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlyAKzalB5w.woff2"
      },
      "400": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3GqXNgp8wxdOdOn58o2JZg.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3GqXNgp8wxdOdOn4Qo2JZg.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3GqXNgp8wxdOdOn4Uo2JZg.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3GqXNgp8wxdOdOn4so2A.woff2"
      },
      "500": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl3gLzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl3gLzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl3gLzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl3gLzalB5w.woff2"
      },
      "600": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl1QMzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl1QMzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl1QMzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOl1QMzalB5w.woff2"
      },
      "700": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlzANzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlzANzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlzANzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlzANzalB5w.woff2"
      },
      "800": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlywOzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlywOzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlywOzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlywOzalB5w.woff2"
      },
      "900": {
        thai: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlwgPzb1B59H-.woff2",
        vietnamese: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlwgPzaZB59H-.woff2",
        "latin-ext": "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlwgPzadB59H-.woff2",
        latin: "https://fonts.gstatic.com/s/trirong/v15/7r3DqXNgp8wxdOdOlwgPzalB5w.woff2"
      }
    }
  }
});
var fontFamily = "Trirong";
var loadFont = (style, options) => {
  return loadFonts(getInfo(), style, options);
};
export {
  loadFont,
  getInfo,
  fontFamily
};
